packages/editor/src/lib/hooks/useReactiveKeyPressed.ts

```diff
@@ -0,0 +1,57 @@
+import { atom } from '@tldraw/state'
+import { useValue } from '@tldraw/state-react'
+import { useEffect } from 'react'
+
+/**
+ * A hook that tracks whether a specific keyboard key is currently pressed.
+ *
+ * Uses signals to efficiently track key states and only trigger re-renders when the tracked keys change.
+ *
+ * @example
+ * ```ts
+ * const HyperlinkButton = ({ url }: { url: string }) => {
+ *   const shiftPressed = useReactiveKeyPressed(['Shift', "CMD"])
+ *
+ *   return (
+ *     <a
+ *       href={url}
+ *       className={classNames('tl-hyperlink-button', {
+ *         'tl-hyperlink-pointer-event__none': shiftPressed,
+ *       })}
+ *       style={{
+ *         pointerEvents: shiftPressed ? 'none' : 'auto',
+ *       }}
+ *     >
+ *       Visit
+ *     </a>
+ *   )
+ * }
+ * ```
+ *
+ * @param keys - Array of key names to track (e.g. `['Shift', 'Meta', 'Escape', 'a']`).
+ * @returns A boolean indicating whether any of the specified keys are currently pressed.
+ *
+ * @public
```

Comment:
```suggestion
 * @internal
```
@SomeHats had a PR a couple of months ago to make Editor.inputs reactive, and I think I'd much rather see us resurrect that idea  than expose a new hook specifically for keys. I think the PR was more or less done ? https://github.com/tldraw/tldraw/pull/5721

no need to bring that back for this PR, but maybe we keep the hook as internal until we have reactive inputs?